packages/core/src/components/ColorPalette/VariantRow.tsx

```diff
@@ -111,6 +112,31 @@ export function VariantRow({
     return () => clearTimeout(copiedTimeout)
   }, [hasCopiedHexCode])
 
+  useEffect(() => {
+    let timer: NodeJS.Timeout | undefined = undefined
+
+    if (timer) clearTimeout(timer)
+
+    if (color !== token.value) {
+      timer = setTimeout(() => {
+        setChanging(false)
+        onUpdateVariant(
+          {
+            ...variant,
+            token: {
+              ...variant.token,
+              value: color,
+            },
+          },
+          isBase
+        )
+      }, 250)
+    }
+    return () => {
+      clearTimeout(timer)
+    }
+  }, [color])
```

Comment:
@Teddarific  and @alexdanilowicz I did this to debounce the updates. So that while the user is changing the slider it won't update. I can try to remove it and see if I can do something else.

Reply:
I don't think debouncing is the right approach. I don't like relying on arbitrary time to fix something. I think we should remove the timers. I'm cool with merging this PR without the timers and we fix the HSL issue separately. 


Want to remove the timers, we get this in a mergeable state, and we go from there? @karanvirsb 

cc @Teddarific

Reply:
Hey @alexdanilowicz if we remove the timer then it will cause a render every time the slider changes.

Reply:
moving convo to: https://mirrorful.slack.com/archives/C04RR2D7C65/p1684188510158349?thread_ts=1684125237.982739&cid=C04RR2D7C65

Reply:
How about using `onChangeEnd`  [ [ref](https://chakra-ui.com/docs/components/slider#getting-the-final-value-when-dragging-the-slider) ]

Reply:
Yeah used onChangeEnd and onChangeStart to tackle the timer. Thanks for the suggestion @Pranav2612000 !